AzureResourceManager/Artifactory/nested/Postgresql_deploy.json

```diff
@@ -91,76 +91,136 @@
+        "addressPrefix": "[variables('subnetDbPrefix')]",
+        "delegations": [
+          {
+            "name": "dlg-Microsoft.DBforPostgreSQL-flexibleServers",
+            "properties": {
+              "serviceName": "Microsoft.DBforPostgreSQL/flexibleServers"
+            }
+          }
+        ],
+        "networkSecurityGroup": {
+          "id": "[resourceId('Microsoft.Network/networkSecurityGroups', parameters('nsg_name'))]"
         }
-      },
+      }
+    },
+    {
+      "type": "Microsoft.Network/privateDnsZones",
+      "apiVersion": "2018-09-01",
+      "name": "[concat(parameters('db_server'), '.private.postgres.database.azure.com')]",
+      "location": "global"
+    },
+    {
+      "type": "Microsoft.Network/privateDnsZones/virtualNetworkLinks",
+      "apiVersion": "2020-06-01",
+      "name": "[concat(parameters('db_server'), '.private.postgres.database.azure.com/vLink')]",
+      "location": "global",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/privateDnsZones/', concat(parameters('db_server'), '.private.postgres.database.azure.com'))]"
+      ],
+      "properties": {
+        "virtualNetwork": {
+          "id": "[resourceId('Microsoft.Network/virtualNetworks/', parameters('virtual_network_name'))]"
+        },
+        "registrationEnabled": false
+      }
+    },
+    {
+      "type": "Microsoft.DBforPostgreSQL/flexibleServers",
+      "apiVersion": "2021-06-01",
+      "name": "[parameters('db_server')]",
+      "location": "[parameters('db_location')]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Network/privateDnsZones/', concat(parameters('db_server'), '.private.postgres.database.azure.com'))]",
+        "[resourceId('Microsoft.Network/privateDnsZones/virtualNetworkLinks/', concat(parameters('db_server'), '.private.postgres.database.azure.com'), 'vLink')]"
+      ],
       "sku": {
-        "name": "[parameters('skuName')]",
-        "tier": "[parameters('skuTier')]",
-        "capacity": "[parameters('skuCapacity')]",
-        "size": "[parameters('skuSizeMB')]",
-        "family": "[parameters('skuFamily')]"
+        "name": "Standard_D2ds_v4",
+        "tier": "GeneralPurpose"
       },
-      "resources": [
-        {
-          "name": "[uniqueString(parameters('db_server'), 'AllowAllWindowsAzureIps' )]",
-          "type": "firewallRules",
-          "apiVersion": "2017-12-01",
-          "location": "[parameters('db_location')]",
-          "dependsOn": [
-            "[resourceId('Microsoft.DBforPostgreSQL/servers', parameters('db_server'))]"
-          ],
-          "properties": {
-            "startIpAddress": "0.0.0.0",
-            "endIpAddress": "0.0.0.0"
-          }
+      "properties": {
+        "administratorLogin": "[parameters('db_user')]",
+        "administratorLoginPassword": "[parameters('db_password')]",
+        "availabilityZone": "",
+        "Backup": {
+          "backupRetentionDays": "7",
```

Comment:
Please, use the values supplied in the parameters. We have it above `[parameters('backupRetentionDays')]`